spec/var-ref.test.sh

```diff
@@ -78,8 +79,9 @@ A_nobrackets=0
 
 #### ${!a[@]-'default'} is illegal
 
-# bash disallows this when a is an array.  We make it an error because [@]
-# implies it's an array.
+# bash disallows this when a is an array.  We originally made it an error
```

Comment:
I'm having trouble with part ...  

I'm not sure if this is the right change -- we are differing from bash in at least some case

And if so, I think the title of the spec test should be updated

Maybe it can be split into two test cases -- when `a` is an array, and when it is not?  Or when it has a single element?

Reply:
> I'm not sure if this is the right change -- we are differing from bash in at least some case

Ah, No. Actually, the statement "bash disallows this when a is an array" is false, and the change in this PR makes osh more consistent with Bash actually. The remaining difference is that osh dies (i.e., `e_die()`) with more than one element, while Bash returns `1` and continues the execution of the subsequent commands.

I'll update the sentence "bash disallows ...".

> And if so, I think the title of the spec test should be updated

Yeah, I think that the title of the spec test doesn't practically reflect the Bash behavior. I'll update it.

> Maybe it can be split into two test cases -- when `a` is an array, and when it is not? Or when it has a single element?

Actually, we already have separate test cases. This existing spec test is for the case where there are more than one element, and the other test case I newly added is the test for the case with a single element.

Reply:
810389985